packages/grafana-data/src/transformations/transformers/insertValues.ts

```diff
@@ -0,0 +1,141 @@
+import { DataFrame, FieldType, Field, ArrayVector } from '../..';
+
+import { map } from 'rxjs/operators';
+
+import { DataTransformerID } from './ids';
+import { DataTransformerInfo } from '../../types/transformations';
+
+const enum InsertValuesMode {
+  Interval = 'interval',
+  Threshold = 'threshold',
+}
+
+const IS_TIME_FIELD = (field: Field) => field.type === FieldType.time;
+const THRESHOLD_HOUR = 3600 * 1e3;
+
+export interface InsertValuesTransformerOptions {
+  mode: InsertValuesMode;
+  // value to insert
+  value: any;
+  // finds field to use for interval or threshold checking
+  isRefField?: (field: Field) => boolean;
+  // minimum threshold required between adjacent non-null/non-undefined values
+  threshold?: number;
+  // should refField's values be scanned to find minimum interval?
+  guessInterval?: boolean;
```

Comment:
is there an use-case for `guessInterval=true` ? it's very cool 👍 , but right now i see no situation where the datasource wants autoi-null-insert, but does not know the `interval`.